frontend/src/App.tsx

```diff
@@ -0,0 +1,111 @@
+import { bind } from 'decko';
+import React, { Component } from 'react';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { connect, DispatchProp } from 'react-redux';
+
+// Bootstrap components
+import Container from 'react-bootstrap/Container';
+
+// CSS and JS
+import 'jquery';
+import 'bootstrap/dist/js/bootstrap.min.js';
+
+import 'font-awesome/css/font-awesome.min.css';
+import 'bootstrap/dist/css/bootstrap.min.css';
+import 'bootswatch/dist/darkly/bootstrap.min.css';
+import './css/App.css';
+
+// Component Includes
+import Navbar from './components/fragments/global/navbar.component';
+import Footer from './components/fragments/global/footer.compontent';
+import Homepage from './components/pages/misc/homepage.component';
+import About from './components/pages/misc/about.component';
+import Profile from './components/pages/user/profile.component';
+
+import { IAppState, IUser } from 'store/reduxState';
+
+// Views
+import { EmotesView } from './views/emotes.view';
+
+interface IAppProps {
+  user?: IUser;
+  error?: Error;
+  authenticated: boolean;
+}
+
+type IAllAppProps =
+  IAppProps &
+  DispatchProp;
+
```

Comment:
The above interfaces are used to "type" the props used by the component.  In this case, I have merged two types into one. This is so the "mapStateToProps" later can map only the properties it needs.

The DispatchProps comes for free with the connect HOF.